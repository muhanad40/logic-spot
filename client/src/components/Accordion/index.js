import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.div`
  background: #F8F8F8;
  padding: 8px 16px;
  border-radius: 4px;
  display: flex;
  justify-content: space-between;
  cursor: pointer;
  margin-bottom: 8px;

  &:after {
    content: '+';
    display: block;
    margin-left: 16px;
    font-size: 20px;
    line-height: 1;
    font-weight: 500;
    color: #5F6367;
  }
`;
  
const Content = styled.div`
  padding: 16px;
  margin-bottom: 12px;
`;

const Accordion = ({ title, content }) => {
  const [showContent, setShowContent] = useState(false);
  const toggle = () => {
    setShowContent(!showContent);
  }

  return (
    <>
      <Title data-test-role="accordion-title" onClick={toggle}>{title}</Title>
      {showContent &&
        <Content data-test-role="accordion-content">{content}</Content>
      }
    </>
  );
}

Accordion.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
};

export default Accordion;