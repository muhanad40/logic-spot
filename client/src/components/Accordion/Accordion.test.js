import React from 'react';
import { FalconClientMock } from '@deity/falcon-client/test-utils';
import { shallow } from 'enzyme';

import Accordion from ".";

describe('Accordion', () => {
  it('should render correctly', () => {
    expect(shallow(<Accordion title="Accordion title..." content="Content..." />)).toMatchSnapshot();
  });

  it('should toggle content visibility when title is clicked', () => {
    const component = shallow(<Accordion title="Test title" content="Test content..." />);

    component.find('[data-test-role="accordion-title"]').simulate('click');
    expect(component.find('[data-test-role="accordion-content"]').exists()).toBe(true);
    component.find('[data-test-role="accordion-title"]').simulate('click');
    expect(component.find('[data-test-role="accordion-content"]').exists()).toBe(false);
  });
});
