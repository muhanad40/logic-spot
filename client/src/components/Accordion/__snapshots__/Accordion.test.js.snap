// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render correctly 1`] = `
<Fragment>
  <styled.div
    data-test-role="accordion-title"
    onClick={[Function]}
  >
    Accordion title...
  </styled.div>
</Fragment>
`;
